performance/base.rb

```diff
@@ -0,0 +1,65 @@
+module Performance
+  class Base
+    def initialize
+      @results = {}
+    end
+
+    def setup; end
+    def teardown; end
+
+    def subject; raise NotImplementedError; end
+
+    def run
+      self.run_process_time
+      self.run_wall_time
+
+      self.print
+    end
+
+    protected
+
+    def run_wall_time
+      self.setup
+
+      RubyProf.measure_mode = RubyProf::WALL_TIME
+
+      @results['Wall Time'] = []
+      (1..5).each do |it|
+        @results['Wall Time'] << RubyProf.profile do
+          self.subject
+        end
+      end
+
+      self.teardown
+    end
+
+    def run_process_time
+      self.setup
+
+      RubyProf.measure_mode = RubyProf::PROCESS_TIME
+
+      @results['Process Time'] = []
+      (1..5).each do |it|
```

Comment:
The gem rails-perftest (https://github.com/rails/rails-perftest) uses ruby-prof and their default number of runs is 5.